### DISCUSSION TITLE
Checking the usage of OpenAI Prompt Caching

### DISCUSSION THREAD

**Discussion by s2terminal** [danny-avila/LibreChat#4334] [Q&A]
I want to check the usage of [OpenAI Prompt Caching](https://openai.com/index/api-prompt-caching/). OpenAI Prompt Caching is applied automatically. I want to check if the cache is enabled for the conversations I had with LibreChat.

I have enabled the `include_usage` option and am checking in the debug log.

`api/app/clients/OpenAIClient.js` (tag: v0.7.5-rc2)

```diff
@@ -1199,6 +1204,7 @@ ${convo}
           .stream({
             ...modelOptions,
             stream: true,
+            stream_options: { include_usage: true },
           })
           .on('abort', () => {
             /* Do nothing here */
```

The log looks like this. I can confirm that the cache is being used by the fact that `prompt_tokens_details.cached_tokens` is being output.

```
2024-10-04T08:11:12.540Z debug: [OpenAIClient] chatCompletion response
{
  object: "chat.completion",
    usage.prompt_tokens: 2455,
    usage.completion_tokens: 70,
    usage.total_tokens: 2525,
    prompt_tokens_details.cached_tokens: 2304,
    completion_tokens_details.reasoning_tokens: 0,
...
```

Is there a better way?

**Comment by danny-avila** [comment]
will be implementing this soon to be handled in transactions
